cdc/sink/codec/canal/canal_json_encoder.go

```diff
@@ -36,16 +36,19 @@ type JSONBatchEncoder struct {
 	// When it is true, canal-json would generate TiDB extension information
 	// which, at the moment, only includes `tidbWaterMarkType` and `_tidb` fields.
 	enableTiDBExtension bool
+	// the symbol separating two lines
+	terminator string
 
 	messages []*common.Message
 }
 
 // newJSONBatchEncoder creates a new JSONBatchEncoder
-func newJSONBatchEncoder(enableTiDBExtension bool) codec.EventBatchEncoder {
+func newJSONBatchEncoder(enableTiDBExtension bool, terminator string) codec.EventBatchEncoder {
```

Comment:
this looks good to me, but as I know, both arguments passed by the config's fields. we can refactor this to pass config as the parameter directly?